zulipterminal/model.py

```diff
@@ -516,6 +516,45 @@ def send_stream_message(self, stream: str, topic: str, content: str) -> bool:
             notify_if_message_sent_outside_narrow(composition, self.controller)
         return message_was_sent
 
+    def prevent_compose_box_in_stream(self, stream_identifier: Any) -> bool:
+        """
+        Identify whether a user is authorised to send message to the given
+        stream or not using stream_post_policy and/or is_announcement_policy
+        and allow/restrict the compose box from opening and accordingly
+        update the footer
+        """
+        if self.stream_dict[stream_identifier].get(
+            "stream_post_policy"
+        ) == 2 or self.stream_dict[stream_identifier].get("is_announcement_only"):
+            # checks whether the user is an admin/owner
+            if (
+                self.get_user_info(self.user_id).get("role") <= 200
+                or self.initial_data.get("is_admin", True)
+                or self.initial_data.get("is_owner", True)
```

Comment:
I'm guessing `get_user_info` could work here, but the type is wrong so mypy is complaining.

Do we need more than the `get_user_info`?

In principle we could track the role of the user who is running the app separately, but we already have the logic in `get_user_info`. That method does currently do a lot of work which we won't use here, however.

Reply:
I too thought of eliminating the use of the function by extracting value of the `role` variable alone. However, going through the changelogs, I noticed that `role` was introduced in Zulip 4.0 (ZFL 59) and the moderator role was added in Zulip 4.0 (ZFL 60), so using `role` alone can cause problems with respect to backwards compatibility. We would also have to use the `is_admin` and `is_owner` attributes to make sure it runs smoothly for earlier versions as well.

One way to do it would be following your suggestion about assigning the user information to a variable. After assigning, we can verify whether the given user exists or not (as the mypy error is due to the `Optional` return type of `get_user_info`). I tried this on my local machine and mypy didn't throw any errors for it.

Reply:
I'm a little wary of duplicating the logic in this new method. That said, let's get it working and tested against some known server responses, and we can refactor next.

Reply:
Considering your suggestion regarding `get_user_info` doing a lot more than what we need here, I considered this logic as an alternative to using the `get_user_info` function within the new method.